src/common/OpenNewWindow.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext } from 'react';
+import {
+  Link, useLocation,
+} from 'react-router-dom';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
+
+/**
+ * Indicates the current window type the extension is running in
+ */
+export enum WindowType {
+  // Smaller extension window
+  EXTENSION,
+  // Full-fledged browser window
+  FULLSCREEN,
+}
+
+export const WindowTypeContext = React.createContext({
+  windowType: WindowType.EXTENSION,
+  setWindowType: (_: WindowType) => {}, /* eslint-disable-line */
+});
+
+const OpenNewWindow = function OpenNewWindow() {
+  // Get global WindowType state
+  const { windowType, setWindowType } = useContext(WindowTypeContext);
+  const location = useLocation();
+
+  // Check search params to see if WindowType changed
+  if (location.search) {
+    const searchParams = new URLSearchParams(location.search);
+    const newWindowType = searchParams.get('windowType');
+    if (newWindowType === 'fullscreen') {
```

Comment:
We have control over the pop-up page in the manifest, so I wonder if it's possible to pass query parameters that way. If it is, I think we should invert this check so that it defaults to the fullscreen view and alleviates the problem of having the button reappear if the user refreshes the page.

Reply:
Because we route through ExistingWallet, I initialize the app's default route to go there with a ?windowType=popup query parameter. Flip it there, and now the app will hide it. After opening in new window, this route isn't used with the query param, so it remains fullscreen even through refreshes.